File: 백준/Silver/9020. 골드바흐의 추측/골드바흐의 추측.js
const fs = require("fs");
const input = fs
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map(Number);

const T = input[0];

function is_prime(n) {
  if (n == 1) return false;
  for (let i = Math.floor(Math.sqrt(n)); i > 1; i--) {
    if (n % i == 0) return false;
  }
  return true;
}

for (let i = 0; i < T; i++) {
  let num = input[i + 1];
  let half = Math.floor(num / 2);
  for (let j = half; j > 1; j--) {
    if (is_prime(j) && is_prime(num - j)) {
      console.log(j, num - j);
      break;
    }
  }
}

File: 백준/Silver/9020. 골드바흐의 추측/골드바흐의 추측.py
import sys
import math

n = int(sys.stdin.readline())
arr = list(int(sys.stdin.readline()) for _ in range(n))

box = [1] * 10001
box[0] = 0
box[1] = 0

for i in range(2, 101):
    j = 2
    if(box[i] == 1):
        while(i * j < 10001):
            box[i * j] = 0      
            j = j + 1

result = []

for i in range(10001):
    if(box[i] == 1):
        result.append(i)
        
for num in arr:
    x = num // 2
    y = num // 2
    while(True): 
        if(x in result):
            if(y in result):
                print(x, y)
                break
        x = x - 1
        y = y + 1
        